modules/sor/sorV2/lib/utils/helpers.ts

```diff
@@ -71,3 +76,46 @@ export function getOutputAmount(paths: PathWithAmount[]): TokenAmount {
     const amounts = paths.map((path) => path.outputAmount);
     return amounts.reduce((a, b) => a.add(b));
 }
+
+export function getHookState(pool: any): HookState | undefined {
```

Comment:
I feel uncomfortable with this `any`, but I think it is what it is 🤷 

@gmbronco - you're the TS magician - any way we can handle this is a safer way? Or is it good enough?

My main issue is with inconsistencies between what we expect and what we get. For example, if hook name is `ExitFee`, we expect it to have a `dynamicData` as attribute. What happens if it's not there for any reason?

Reply:
wouldn't PrismaPoolAndHookWithDynamic work? as defined in parent functions.

Reply:
Changing it to `PrismaPoolAndHookWithDynamic` introduces build errors

```
modules/sor/sorV2/lib/utils/helpers.ts:87:19 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

87     if (pool.hook.name === 'ExitFee') {
                     ~~~~

modules/sor/sorV2/lib/utils/helpers.ts:89:39 - error TS2339: Property 'dynamicData' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'dynamicData' does not exist on type 'string'.

89         const dynamicData = pool.hook.dynamicData as { removeLiquidityFeePercentage: string };
                                         ~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:98:19 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

98     if (pool.hook.name === 'DirectionalFee') {
                     ~~~~

modules/sor/sorV2/lib/utils/helpers.ts:103:19 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

103     if (pool.hook.name === 'StableSurge') {
                      ~~~~

modules/sor/sorV2/lib/utils/helpers.ts:105:18 - error TS18047: 'pool.typeData' is possibly 'null'.

105             amp: pool.typeData.amp,
                     ~~~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:105:32 - error TS2339: Property 'amp' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'amp' does not exist on type 'string'.

105             amp: pool.typeData.amp,
                                   ~~~

modules/sor/sorV2/lib/utils/helpers.ts:106:50 - error TS18047: 'pool.dynamicData' is possibly 'null'.

106             surgeThresholdPercentage: parseEther(pool.dynamicData.surgeThresholdPercentage),
                                                     ~~~~~~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:106:67 - error TS2339: Property 'surgeThresholdPercentage' does not exist on type 'PrismaPoolDynamicData'.

106             surgeThresholdPercentage: parseEther(pool.dynamicData.surgeThresholdPercentage),
                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:110:34 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

110     throw new Error(`${pool.hook.name} hook not implemented`);
                                     ~~~~


Found 9 errors in the same file, starting at: modules/sor/sorV2/lib/utils/helpers.ts:87
```

Reply:
I'll look more into how I can address the `any`.

Reply:
Simply replacing the type with `PrismaPoolAndHookWithDynamic` introduces build errors 

```
modules/sor/sorV2/lib/utils/helpers.ts:87:19 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

87     if (pool.hook.name === 'ExitFee') {
                     ~~~~

modules/sor/sorV2/lib/utils/helpers.ts:89:39 - error TS2339: Property 'dynamicData' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'dynamicData' does not exist on type 'string'.

89         const dynamicData = pool.hook.dynamicData as { removeLiquidityFeePercentage: string };
                                         ~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:98:19 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

98     if (pool.hook.name === 'DirectionalFee') {
                     ~~~~

modules/sor/sorV2/lib/utils/helpers.ts:103:19 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

103     if (pool.hook.name === 'StableSurge') {
                      ~~~~

modules/sor/sorV2/lib/utils/helpers.ts:105:18 - error TS18047: 'pool.typeData' is possibly 'null'.

105             amp: pool.typeData.amp,
                     ~~~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:105:32 - error TS2339: Property 'amp' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'amp' does not exist on type 'string'.

105             amp: pool.typeData.amp,
                                   ~~~

modules/sor/sorV2/lib/utils/helpers.ts:106:50 - error TS18047: 'pool.dynamicData' is possibly 'null'.

106             surgeThresholdPercentage: parseEther(pool.dynamicData.surgeThresholdPercentage),
                                                     ~~~~~~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:106:67 - error TS2339: Property 'surgeThresholdPercentage' does not exist on type 'PrismaPoolDynamicData'.

106             surgeThresholdPercentage: parseEther(pool.dynamicData.surgeThresholdPercentage),
                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~

modules/sor/sorV2/lib/utils/helpers.ts:110:34 - error TS2339: Property 'name' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'name' does not exist on type 'string'.

110     throw new Error(`${pool.hook.name} hook not implemented`);
                                     ~~~~


Found 9 errors in the same file, starting at: modules/sor/sorV2/lib/utils/helpers.ts:87
```

Reply:
Yeah, I don't think it makes sense to use `PrismaPoolAndHookWithDynamic`.
Here we're handling the `hook`  attribute from `PrismaPool`, which is stored as a Json, meaning it can be essentially anything.
I think we should be ok to go with `any` for now and refactor in the future in case we find a better way to handle it @mkflow27